### ISSUE TITLE
🐛 Error from Rust Analyzer with example axum web project 🚨 "proc macro `main` not expanded: cannot find proc-macro-srv"

### ISSUE THREAD

**Issue by pcgeek86** [rust-lang/rust-analyzer#15531]
### Summary

I'm creating an axum web server example on Windows 11, and running into an issue with Rust Analyzer. The project builds and runs just fine, but Rust Analyzer still throws the error:

> proc macro `main` not expanded: cannot find proc-macro-srv, the workspace `C:\git\rust-web` is missing a sysroot

<img width="1408" alt="image" src="https://github.com/rust-lang/rust-analyzer/assets/466713/bbc79be6-36d8-4ea9-815b-7aeb9bdf5f57">

```
PS C:\git\rust-web> cargo build
   Compiling scopeguard v1.1.0
   Compiling smallvec v1.11.0
   Compiling num_cpus v1.16.0
   Compiling windows-sys v0.48.0
   Compiling lock_api v0.4.10
   Compiling parking_lot_core v0.9.8
   Compiling parking_lot v0.12.1
   Compiling socket2 v0.5.3
   Compiling mio v0.8.8
   Compiling tokio v1.32.0
   Compiling tower v0.4.13
   Compiling hyper v0.14.27
   Compiling axum v0.6.20
   Compiling rust-web v0.1.0 (C:\git\rust-web)
    Finished dev [unoptimized + debuginfo] target(s) in 12.21s
```

### Repro Steps

* `mkdir rust-web; cd rust-web; cargo init`
* `cargo add axum`
* Update `tokio` in `Cargo.toml` to this value: `tokio = { version = "1.32.0", features = ["full"] }`
* Use `main.rs` below
* `cargo run`

```rust
// main.rs
use axum::{
    routing::get,
    Router,
};


#[tokio::main]
async fn main() {
    // build our application with a single route
    let app = Router::new().route("/", get(|| async { "Hello, World!" }));

    // run it with hyper on localhost:3000
    axum::Server::bind(&"0.0.0.0:3000".parse().unwrap())
        .serve(app.into_make_service())
        .await
        .unwrap();
}
```

**rust-analyzer version**: `v0.3.1641`

**rustc version**: `rustc 1.72.0 (5680fa18f 2023-08-23)`

**relevant settings**: 

VSCode 1.81.1 6c3e3dba23e8fadc360aed75ce363ba185c49794 x64

**Comment by davidbarsky** [comment]
How have you installed Rust? You might be missing some components, as https://stackoverflow.com/questions/76171390/proc-macro-main-not-expanded-rust-analyzer-not-spawning-server appears to be quite similar to your issue.

**Comment by pcgeek86** [comment]
> How have you installed Rust? 

I just use my package manager to install / update the latest Rust toolchain. Currently on 1.72.0.

```
scoop install rust
```

**Comment by davidbarsky** [comment]
@pcgeek86 I don't exactly know how scoop packages Rust, but knowing what I know about _other_ package managers, it's entirely possible that it is missing `proc-macro-srv` component that is installed as when using rustup. Can you try that instead?

**Comment by pcgeek86** [comment]
> possible that it is missing proc-macro-srv component

Is there any way to verify this independently? ie. Rust Analyzer expects `proc-macro-srv.exe` to be on `%PATH%` or something?

**Comment by davidbarsky** [comment]
> Is there any way to verify this independently? ie. Rust Analyzer expects `proc-macro-srv.exe` to be on `%PATH%` or something?

I'm not sure what sort of independent verification you're looking for, but rust-analyzer looks for the `proc-macro-srv` in your sysroot: https://github.com/rust-lang/rust-analyzer/blob/8edd81f166cffeb727aa7b1405a9eeb22641762b/crates/project-model/src/workspace.rs#L443 If `scoop` is not bundling the full sysroot with `scoop install rust`, then that would explain this error. For context, on my Mac, my current sysroot is located at `~/.rustup/toolchains/stable-aarch64-apple-darwin` and the procedural macro server is present at `~/.rustup/toolchains/stable-aarch64-apple-darwin/libexec/rust-analyzer-proc-macro-srv`.

I also just downloaded the stable `aarch64-unknown-linux-musl` toolchain from https://forge.rust-lang.org/infra/other-installation-methods.html#standalone-installers, and while this will obviously not on Windows, I noticed that `proc-macro-srv` is in the `rustc/libexec/` folder (instead of just `libexec`), so rust-analyzer might not be able to find it.

In general, I *really* recommend using `rustup` if you're developing Rust code. OS package managers have historically not provided an experience optimized for that purpose whereas Rustup has.
